compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2432,6 +2433,15 @@ private void validateDefaultable(BLangRecordTypeNode recordTypeNode) {
         }
     }
 
+    private void validateOptionalNeverTypedField(BLangRecordTypeNode recordTypeNode) {
+        // Never type is only allowed in an optional field in a record
+        for (BLangSimpleVariable field : recordTypeNode.fields) {
+            if (field.type.tag == TypeTags.NEVER && !field.flagSet.contains(Flag.OPTIONAL)) {
+                dlog.error(field.pos, DiagnosticCode.NEVER_TYPE_NOT_ALLOWED_FOR_REQUIRED_FIELDS, field.name.value);
```

Comment:
This semantic rule about only optional fields being allowed to be `never` type fields is not included in the spec right? It's something we discussed? If so, will have to bring this up with James and add it to the spec

Reply:
We discussed about it . But, it is not included in the spec.

Reply:
Can you put a comment in the spec issue explaining the scenario and the conclusion we came to?